### ISSUE TITLE
Implement merge

### ISSUE THREAD

**Issue by jisantuc** [jisantuc/interval-index#4]
For combining two trees.

What to do with intervals "present in both"? I don't have a notion of id'ed intervals yet. For now, split `merge` and `mergeDeduplicate`, where `mergeDeduplicate` has an... `Eq` constraint on `a` (I think it's `Eq` for `Set` at least)

**Comment by jisantuc** [comment]
Closed initially by #9, if I want to work on performance I'll open a new issue.

**Comment by jisantuc** [comment]
"Make performance better" is unblocked by adding benchmarks in #13 / #12 -- `merge` is not surprisingly _pretty rough_:

![image](https://github.com/user-attachments/assets/b4e801c0-30ba-4ad1-b3cd-c83f2bfe0b24)

Even for small inputs, the intrinsic slowness of how it works right now (`foldl' insert`) makes it more expensive to merge a 7 element interval index with a 3 element interval index than it is to construct a 10 element interval index. That's weird! I already know so much stuff about the 7/3 element interval indices! `fromList` timings should basically be an upper bound for `merge`, since in `fromList` I know literally nothing about the data, while in `merge` both operands are already indexed.
